airbyte-integrations/bases/base-java-s3/src/main/java/io/airbyte/integrations/destination/s3/jsonl/S3JsonlFormatConfig.java

```diff
@@ -6,35 +6,74 @@
 
 import static io.airbyte.integrations.destination.s3.S3DestinationConstants.COMPRESSION_ARG_NAME;
 import static io.airbyte.integrations.destination.s3.S3DestinationConstants.DEFAULT_COMPRESSION_TYPE;
-import static io.airbyte.integrations.destination.s3.S3DestinationConstants.FLATTEN_DATA;
 
+import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.databind.JsonNode;
 import io.airbyte.integrations.destination.s3.S3Format;
 import io.airbyte.integrations.destination.s3.S3FormatConfig;
 import io.airbyte.integrations.destination.s3.util.CompressionType;
 import io.airbyte.integrations.destination.s3.util.CompressionTypeHelper;
+
 import java.util.Objects;
 
 public class S3JsonlFormatConfig implements S3FormatConfig {
 
   public static final String JSONL_SUFFIX = ".jsonl";
 
-  private final CompressionType compressionType;
+  public enum Flattening {
+
+    // These values must match the format / jsonl_flattening enum values in spec.json.
+    NO("No flattening"),
+    ROOT_LEVEL("Root level flattening");
+
+    private final String value;
+
+    Flattening(final String value) {
+      this.value = value;
+    }
 
-  private final boolean flattenData;
+    @JsonCreator
+    public static S3JsonlFormatConfig.Flattening fromValue(final String value) {
```

Comment:
I think theres maybe an awkward thing happening here where the Sentence case formatting of the enum values makes it hard to mutate an input string to compare. There is a built in [valueOf](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Enum.html#valueOf(java.lang.Class,java.lang.String)) you could leverage here which does almost the same thing. If the enum values could be changed to lowercase  then it would make it easier to use the builtin method like so 
```
S3JsonlFormatConfig.Flattening.valueOf(value.toLowerCase());
```